### PR TITLE
Index all smart contract results 

### PR DESCRIPTION
**Pull request by miiu96**
Instead of indexing just one smart contract result that belongs to a transactions now are indexed all.

### CODE REVIEW

File: core/indexer/processTransactions.go [hunk 1]
```diff
@@ -120,20 +120,30 @@ func (tdp *txDatabaseProcessor) prepareTransactionsForDatabase(
 
 func (tdp *txDatabaseProcessor) addScResultInfoInTx(scr *smartContractResult.SmartContractResult, tx *Transaction) *Transaction {
 	dbScResult := tdp.commonProcessor.convertScResultInDatabaseScr(scr)
-	if tx.Sender != dbScResult.Receiver || len(dbScResult.Data) == 0 {
-		return tx
-	}
+	// now all smart contract results that belongs to a transaction are indexed
+	//if tx.Sender != dbScResult.Receiver || len(dbScResult.Data) == 0 {
```
Comment (sasurobert) [inline core/indexer/processTransactions.go:124]: no commented code

File: core/indexer/processTransactions.go [hunk 2]
```diff
@@ -120,20 +120,30 @@ func (tdp *txDatabaseProcessor) prepareTransactionsForDatabase(
...
-	if dbScResult.GasLimit != 0 && dbScResult.Value != "0" {
+	if isSCWithGasUsed(dbScResult, tx) {
 		gasUsed := tx.GasLimit - scr.GasLimit
 		tx.GasUsed = gasUsed
 	}
 
 	return tx
 }
 
+func isSCWithGasUsed(dbScResult ScResult, tx *Transaction) bool {
```
Comment (sasurobert) [inline core/indexer/processTransactions.go:138]: no need for this function

Comment (sasurobert) [reply core/indexer/processTransactions.go:138]: check if prevTxHash == originalTxHash == txHash

File: core/indexer/processTransactions.go (hunk 1 above)
Comment (miiu96) [reply core/indexer/processTransactions.go]: done.

File: core/indexer/processTransactions.go (hunk 2 above)
Comment (miiu96) [reply core/indexer/processTransactions.go]: done.

**Pull request merged**